KachnaOnline.Business/Facades/UserFacade.cs

```diff
@@ -0,0 +1,99 @@
+// UserFacade.cs
+// Author: František Nečas
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoMapper;
+using KachnaOnline.Business.Services.Abstractions;
+using KachnaOnline.Dto.Users;
+using KachnaOnline.Business.Exceptions;
+using KachnaOnline.Business.Exceptions.Roles;
+using KachnaOnline.Business.Models.Users;
+
+namespace KachnaOnline.Business.Facades
+{
+    public class UserFacade
+    {
+        private readonly IMapper _mapper;
+        private readonly IUserService _userService;
+
+        public UserFacade(IUserService userService, IMapper mapper)
+        {
+            _mapper = mapper;
+            _userService = userService;
+        }
+
+        /// <summary>
+        /// Returns an enumerable of all users.
+        /// </summary>
+        /// <returns>An enumerable of all <see cref="UserDto"/>.</returns>
+        public async Task<IEnumerable<UserDto>> GetUsers()
+        {
+            var users = new List<UserDto>();
+            foreach (var user in await _userService.GetUsers())
+            {
+                var userDto = _mapper.Map<UserDto>(user);
+                userDto.Roles = (await _userService.GetUserRoles(userDto.Id)).ToArray();
+                users.Add(userDto);
```

Comment:
This must be changed to provide correct information about manually and non-manually assigned roles (see the long comment below).

Reply:
+1, however I'd keep the output of `/users/me` as is (just array of role names). I will add another DTO.